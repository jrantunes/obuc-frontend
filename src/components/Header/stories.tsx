import Header from ".";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";

export default {
  title: "Header",
  component: Header
  //args: {}
} as ComponentMeta<typeof Header>;

export const Default: ComponentStory<typeof Header> = () => (
  <BrowserRouter>
    <Header />
  </BrowserRouter>
);
