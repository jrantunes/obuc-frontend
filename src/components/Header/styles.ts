import styled, { css } from "styled-components";

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: ${theme.spacings.xxsmall} 0;
    max-width: ${theme.grid.container};
    margin: 0 auto;

    img {
      width: 9.6rem;
      height: 9.6rem;
    }

    a {
      text-decoration: none;

      & + a {
        margin-left: ${theme.spacings.small};
      }
    }
  `}
`;
