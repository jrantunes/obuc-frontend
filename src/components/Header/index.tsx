import * as S from "./styles";

import logoImg from "assets/images/logo.png";
import Button from "components/Button";
import { Link, useLocation } from "react-router-dom";

const Header = () => {
  const { pathname } = useLocation();

  return (
    <S.Wrapper>
      <img src={logoImg} alt="Logo da confeitaria dona maria" />
      <div>
        <Link to="/">
          <Button minimal>Vagas</Button>
        </Link>
        {pathname !== "/vagas/novo" && (
          <Link to="/vagas/novo">
            <Button color="secondary" size="small">
              Nova vaga
            </Button>
          </Link>
        )}
      </div>
    </S.Wrapper>
  );
};

export default Header;
