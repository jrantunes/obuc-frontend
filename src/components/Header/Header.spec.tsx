/* eslint-disable @typescript-eslint/ban-ts-comment */
import { screen } from "@testing-library/react";
import { useLocation } from "react-router-dom";
import { renderWithTheme } from "utils/tests/helpers";

import Header from ".";

jest.mock("react-router-dom", () => {
  return {
    //@ts-ignore
    Link: ({ children }) => children,
    useLocation: jest.fn()
  };
});

describe("<Header />", () => {
  it("should render the header", () => {
    (useLocation as jest.Mock).mockReturnValue({
      pathname: "/"
    });

    const { container } = renderWithTheme(<Header />);

    expect(
      screen.getByRole("img", { name: /logo da confeitaria dona maria/i })
    ).toBeInTheDocument();

    expect(screen.getByRole("button", { name: /vagas/i })).toBeInTheDocument();

    expect(
      screen.getByRole("button", { name: /nova vaga/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the header without the new job button", () => {
    (useLocation as jest.Mock).mockReturnValue({
      pathname: "/vagas/novo"
    });

    renderWithTheme(<Header />);

    expect(
      screen.getByRole("img", { name: /logo da confeitaria dona maria/i })
    ).toBeInTheDocument();

    expect(screen.getByRole("button", { name: /vagas/i })).toBeInTheDocument();

    expect(
      screen.queryByRole("button", { name: /nova vaga/i })
    ).not.toBeInTheDocument();
  });
});
