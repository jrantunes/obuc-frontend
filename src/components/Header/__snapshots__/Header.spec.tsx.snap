// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.8rem 0;
  max-width: 130rem;
  margin: 0 auto;
}

.c0 img {
  width: 9.6rem;
  height: 9.6rem;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a + a {
  margin-left: 2.4rem;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #5ce1e6;
  color: #FFFFFF;
  font-family: Roboto,--apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 400;
  border: 0;
  border-radius: 0.4rem;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  height: 3.6rem;
  font-size: 1.8rem;
  padding: 0.8rem 4.8rem;
  background: none;
  color: #5ce1e6;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  padding: 0;
  height: auto;
}

.c1:hover {
  background: none;
}

.c1:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c1:disabled:hover {
  background: #5ce1e6;
}

.c1 svg {
  width: 2.4rem;
  height: 2.4rem;
}

.c1:hover {
  color: #39dae1;
}

.c1:disabled:hover {
  background: none;
  color: #5ce1e6;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #e55175;
  color: #FFFFFF;
  font-family: Roboto,--apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 400;
  border: 0;
  border-radius: 0.4rem;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  height: 3rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c2:hover {
  background: #e23b63;
}

.c2:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c2:disabled:hover {
  background: #e55175;
}

<header
  class="c0"
>
  <img
    alt="Logo da confeitaria dona maria"
    src="logo.png"
  />
  <div>
    <button
      class="c1"
      color="primary"
    >
      <span>
        Vagas
      </span>
    </button>
    <button
      class="c2"
      color="secondary"
    >
      <span>
        Nova vaga
      </span>
    </button>
  </div>
</header>
`;
